lab_2_tokenize_by_bpe/main.py

```diff
@@ -69,6 +154,28 @@ def get_vocabulary(
     :param unknown_token: a token to signify an unknown token
     :return: dictionary in the form of <token: identifier>
     """
+    if not (
+        isinstance(word_frequencies, dict) and
+        isinstance(unknown_token, str)
+    ):
+        return None
+
+    identificators = {}
+    all_tokens = [unknown_token]
+
+    for tuple_word in word_frequencies:
+        for word in tuple_word:
+            all_tokens.append(word)
+            prepared_word = prepare_word(word, None, None)
```

Comment:
For what reason do you make a call to `prepare_word`? It is used to turn raw text into encoded, here you have an already processed text.

Reply:
I thought it was necessary, now I got it. It is fixed